app/src/main/java/com/libs/globaldata/ModelObject.java

```diff
@@ -0,0 +1,43 @@
+package com.libs.globaldata;
+
+import android.content.Context;
+
+import com.libs.facerecognition.NeuralModel;
+import com.libs.facerecognition.models.FaceNetModel;
+import com.libs.globaldata.userdatabase.UserDatabase;
+
+public class ModelObject {
+    public final NeuralModel neuralModel;
+    public final UserDatabase userDatabase;
+
+    /**
+     * ModelObject constructor. Depending on the type, acquires proper neural network model.
+     * Always creates new database with the unique filename assigned.
+     *
+     * @param modelType - type of the neural network
+     * @param context   - app/activity context
+     */
+    ModelObject(ModelType modelType, Context context) {
+        switch (modelType) {
+            case FACENET_USER:

```

Comment:
same things hear, it should be much more generic, base on name. Now when i want to add new model i have to do it in so many places and duplicate code...

Reply:
+1